### PR TITLE
Set up DATABASE Resolves se701g2/Doto#37

### PR DESCRIPTION
**Pull request by lucycy98**
Resolves se701g2/Doto#37

- hosted db on mongodb atlas, created account using assigned email/password
- created basic 'sample' schema for generic user data
- created connection with mongoose
- added basic routing for testing purposes

### CODE REVIEW

File: doto-backend/index.js [hunk 1]
```diff
@@ -4,12 +4,30 @@ const cors = require('cors')
 const app = express()
 const apiPort = 3000
 
+// Mongoose connection
+const mongoose = require('mongoose');
+const connectionString = "mongodb+srv://701Group2:SoftwareEngineering701Group2@cluster0-ahuai.mongodb.net";
```
Comment (tigerj2) [inline doto-backend/index.js]: Can we use an env for this? So that we can mask the username and password.

Comment (lucycy98) [reply doto-backend/index.js]: yeah, done :)

**Comment by lucycy98** [comment]
@daltonlim added the tests, but not sure how I can rename the branch without deleting this PR

File: doto-backend/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
     "description": "",
     "main": "index.js",
     "scripts": {
-        "test": "mocha",
+        "test": "echo \"Error: no test specified\" && exit 0",
         "start": "node ./src/index.js"
```
Comment (tigerj2) [inline doto-backend/package.json]: I think you can change this line to:

"start": "nodemon index.js"

File: doto-backend/index.js [hunk 3]
```diff
@@ -4,12 +4,30 @@ const cors = require('cors')
...
+db.once('open', function(){
+    console.log("Connected to MongoDB.");
+});
+db.on('error', function(){
+    console.log(err);
+});
+
 app.use(bodyParser.urlencoded({ extended: true }))
 app.use(cors())
 app.use(bodyParser.json())
```
Comment (allenwang968) [inline doto-backend/index.js]: Looks good Lucy. But now express has a builtin body parser which you can call the same way you're currently doing, but replace the "bodyParser" bits with "express" instead. So this means we don't have to import the body parser dependency either.

**Comment by lucycy98** [comment]
created new PR se701g2/Doto/pull/47

**Pull request closed without merging**